src/pkg/sources/remote.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The UDS Authors
+
+// Package sources contains Zarf packager sources
+package sources
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	zarfTypes "github.com/defenseunicorns/zarf/src/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content/file"
+
+	"github.com/defenseunicorns/uds-cli/src/config"
+)
+
+// RemoteBundle is a package source for remote bundles that implements Zarf's packager.PackageSource
+type RemoteBundle struct {
+	PkgName        string
+	PkgOpts        *zarfTypes.ZarfPackageOptions
+	PkgManifestSHA string
+	TmpDir         string
+	Remote         *oci.OrasRemote
+}
+
+// LoadPackage loads a Zarf package from a remote bundle
+func (r *RemoteBundle) LoadPackage(dst *layout.PackagePaths, unarchiveAll bool) error {
+	packageSpinner := message.NewProgressSpinner("Loading bundled Zarf package: %s", r.PkgName)
+	defer packageSpinner.Stop()
+	err := r.downloadPkgFromRemoteBundle()
+	if err != nil {
+		return err
+	}
+
+	var files []string
+	var pkg zarfTypes.ZarfPackage
+	err = filepath.Walk(dst.Base, func(path string, fileInfo fs.FileInfo, err error) error {
+		if err != nil {
+			return nil
+		}
+		if fileInfo.IsDir() {
+			return nil
+		}
+		newString := strings.ReplaceAll(path, dst.Base+"/", "")
+		files = append(files, newString)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	if err = utils.ReadYaml(dst.ZarfYAML, &pkg); err != nil {
+		return err
+	}
+
+	dst.SetFromPaths(files)
```

Comment:
this should be using `dst.SetFromLayers`, not `dst.SetFromPaths`

Reply:
Good call, removed the Walk function and returned the proper layers from `downloadPkgFromRemoteBundle`